data/src/apiKey/java/org/cryptomator/data/cloud/onedrive/OnedriveImpl.kt

```diff
@@ -229,12 +235,36 @@ internal class OnedriveImpl(private val cloud: OnedriveCloud, private val client
 						.whenComplete { driveItem, error ->
 							run {
 								if (error == null) {
-									progressAware.onProgress(Progress.completed(UploadState.upload(file)))
-									result.complete(driveItem)
-									cacheNodeInfo(file, driveItem)
-								} else {
+
+									val diffItem = DriveItem()
+									diffItem.fileSystemInfo = FileSystemInfo()
+									if(data.modifiedDate(context).isPresent){
+										diffItem.fileSystemInfo!!.lastModifiedDateTime = OffsetDateTime.ofInstant(data.modifiedDate(context).get().toInstant(), ZoneId.systemDefault())
+									}
+									else{
+										diffItem.fileSystemInfo!!.lastModifiedDateTime = OffsetDateTime.ofInstant(Date().toInstant(), ZoneId.systemDefault())
+									}
```

Comment:
If I'm not wrong this is the default behavior, the server sets the file modification date to now so there is no need to execute an extra request if we can not retrieve the actual modification date.